ds.go

```diff
@@ -523,7 +520,17 @@ func decryptCredential(cred *Credential, encContext *EncryptionContextValue) (*D
 	}
 
 	dk, err := DecryptDataKey(wrappedKey, encContext)
-
+	if awsErr, ok := err.(awserr.Error); ok {
```

Comment:
I've redone as switch/case - I agree should have been that way :)

Ideally I feel it should be unicreds errors not AWS errors being propagated.  However, I didn't want to presume the change and also I haven't looked at the dependencies on AWS errors in the test mocks.